### ISSUE TITLE
Bug in Leaf.copy regarding mismatched input of createparents and recursive parameter

### ISSUE THREAD

**Issue by KoStehner** [PyTables/PyTables#1125]
In `tables/leaf.py` in the method `Leaf.copy` we do have the call (lines 697-698) 
~~~
return self._f_copy(newparent, newname, overwrite, createparents, **kwargs)
~~~

The 'self._f_copy' method is implemented in the 'Node' class in 'tables/node.py`. The signature of 'Node._f_copy' is as follows (lines 709-711):
~~~
def _f_copy(self,
            newparent: Union["Group", str, None]=None,
            newname: Optional[str]=None,
            overwrite: bool=False,
            recursive: bool=False,
            createparents: bool=False,
            **kwargs) -> "Node":
~~~

We can see that the first three inputs (`newparent`, `newname`, `overwrite`) do match their respective parameter in the `Node._f_copy_`method.
However, then we do have a `createparents` input that maps to the `recursive` parameter.
This is most likely not desired.

An easy fix would be to promote the `createparents` input __from a positional one__ 
~~~
return self._f_copy(newparent, newname, overwrite, createparents, **kwargs)
~~~
__to a keyword one__
~~~
return self._f_copy(newparent, newname, overwrite, createparents=createparents, **kwargs)
~~~

However, it should be checked if everything is still working as expected.

**Comment by KoStehner** [comment]
Further, Python itself does provide a mechanism to prevent such mishaps by defining keyword-only arguments.
See https://peps.python.org/pep-3102/ and https://docs.python.org/3/glossary.html#term-parameter

With that we could define 
~~~
def _f_copy(self,
            newparent: Union["Group", str, None]=None,
            newname: Optional[str]=None,
            *,
            overwrite: bool=False,
            recursive: bool=False,
            createparents: bool=False,
            **kwargs) -> "Node":
~~~

With that, `newparent` and `newname` can be set as positional or keyword argument, but `overwrite`, `recursive` and `createparents` will be keyword-only.

This may especially interesting for boolean arguments.
However, it is also a design choice of the project whether it should be used or not.

**Comment by avalentino** [comment]
Hi @KoStehner 
thanks for reporting.
I would appreciate if you could submit a PR for this issue. 
Our test suite is quite comprehensive and evident issues resulting from the change could be spotted just running the test in CI.
After that, of course, I will do additional check to be sure that everything works properly and maybe to add a regression test.

**Comment by avalentino** [comment]
Closed in #1127
